### ISSUE TITLE
Zombie processes created on Kubernetes

### ISSUE THREAD

**Issue by swhitman0** [dmtcp/dmtcp#731]
Hello,

We're using DMTCP version 2.5.2. Our application checkpoints itself by calling dmtcp_checkpoint(). We noticed that zombie processes are created when running in Kubernetes. The program is called storm_model.exe. Here is a short listing from ps:
```
root     14119  5762  0 01:36 ?        00:00:00 [DMTCP:storm_mod] <defunct>
root     14126 10758  0 01:36 ?        00:00:00 [DMTCP:storm_mod] <defunct>
root     14313  5762  0 01:36 ?        00:00:00 [DMTCP:storm_mod] <defunct>
root     14314 10888  0 01:36 ?        00:00:00 [DMTCP:storm_mod] <defunct>
root     14315 10641  0 01:36 ?        00:00:00 [DMTCP:storm_mod] <defunct>
```
We added the --no-gzip flag to dmtcp_launch which prevented gzip zombies, but we are still plagued by the DMTCP zombies.
Any idea on how to resolve this?

Thanks

**Comment by gc00** [comment]
Hi,
    Can you clarify a little?  Are you saying that no
checkpoint is created, and the running process turns into
a zombie process?
    Also, have you tried testing outside of Kubernetes in
standard Linux?
    Is there a version of your code that we could test locally
at our site?

Thanks,
- Gene

----- Original Message -----

**Comment by swhitman0** [comment]
DMTCP worked fine. It created checkpoints. No zombies were created with standard Linux. It was not the running process that turns into a zombie, I believe it was the process that communicates with the coordinator. 

I think it has to do with the way processes are handled in a kubernetes pod. I used python to spawn the process. I fixed the problem by iterating over all of the child processes of the current process and waited for them to end. Since they were zombies the wait returned immediately.
